File: Contests/16/C.c
#include <stdio.h>
#include <string.h>

int main() {
    int t;
    scanf("%d", &t);  // Read the number of test cases
    
    while (t--) {
        char s[300001];
        scanf("%s", s);  // Read the pattern string
        
        char result[300001];
        strcpy(result, s);
        char current = '0';
        
        for (int i = 0; i < strlen(s); i++) {
            if (s[i] == '?') {
                result[i] = current;
            } else {
                current = s[i];
            }
        }
        
        printf("%s\n", result);
    }
    
    return 0;
}

File: Contests/16/C.py
t = int(input())

for _ in range(t):
    s = input()
    result = s
    current = '0'
    for i in range(len(s)):
        if s[i] == '?':
            result = result[:i] + current + result[i+1:]
        else:
            current = s[i]
    
    print(result)
